### PR TITLE
Fixed RIDER-44641. Add unit tests for Rider module

### PR DESCRIPTION
**Pull request by sdubov**
- Add Unit tests for Rider Helper, Validator and other Utils methods
- Add @TestOnly APIs for Azure MVP objects to verify function apps, SQL database and SQL Servers existence on run config validation logic
- Escape all single quotes in localized messages
- Fix issue in vatidation logic for SQL Database
- Update build scripts to run tests on build
- Remove unused code

### CODE REVIEW

**Comment by sdubov** [comment]
@maartenba, please ignore TODOs in a source code outside Mock objects. This commit was too huge to cover all that :) I've exposed some new APIs for testing inside `Utils`. We could cover it in a different ways if you have any issues about this approach.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle [hunk 1]
```diff
@@ -39,5 +39,5 @@ test {
 }
 
 publishPlugin {
-    channels publishChanel
+    channels publish_chanel
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle]: Are we using publish on the build server?

**Review by maartenba — APPROVED** [review]
One question, otherwise LGTM! :shipit:

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle (hunk 1 above)
Comment (sdubov) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle]: My previous intentions. Dropped. We need to make a build config for publishing - https://youtrack.jetbrains.com/issue/RIDER-44653

**Review by maartenba — APPROVED** [review]

**Pull request merged**